### ISSUE TITLE
Intermittent test failure in /lib/es/tests/test_commands.py:test_reindexation_starting_from_one_addon()

### ISSUE THREAD

**Issue by bobsilverberg** [mozilla/addons-server#14467]
I have seen this at least a couple of times on recent PR Travis runs:

```
=================================== FAILURES ===================================

__________ TestIndexCommand.test_reindexation_starting_from_one_addon __________

src/olympia/lib/es/tests/test_commands.py:165: in test_reindexation_starting_from_one_addon

    self._test_reindexation()

        self       = <olympia.lib.es.tests.test_commands.TestIndexCommand testMethod=test_reindexation_starting_from_one_addon>

src/olympia/lib/es/tests/test_commands.py:133: in _test_reindexation

    self.check_results(self.expected)

        ReindexThread = <class 'olympia.lib.es.tests.test_commands.TestIndexCommand._test_reindexation.<locals>.ReindexThread'>

        old_addons_count = 1

        old_indices = [('test_amo_addons-20200602162433', 'test_amo_addons'),

 ('test_amo_addons_stats_download_counts-20200602162433',

  'test_amo_addons_stats_download_counts'),

 ('test_amo_addons_stats_update_counts-20200602162433',

  'test_amo_addons_stats_update_counts')]

        self       = <olympia.lib.es.tests.test_commands.TestIndexCommand testMethod=test_reindexation_starting_from_one_addon>

        t          = <ReindexThread(Thread-1, stopped 140473940182784)>

        wipe       = False

src/olympia/lib/es/tests/test_commands.py:79: in check_results

    assert addon.pk in got, '%s is not in %s' % (addon.pk, got)

E   AssertionError: 1 is not in []

E   assert 1 in []

E    +  where 1 = <Addon: 1: Addôn 1935673f1f1e43a6831543b946c41cd5>.pk

        addon      = <Addon: 1: Addôn 1935673f1f1e43a6831543b946c41cd5>

        expected   = [<Addon: 1: Addôn 1935673f1f1e43a6831543b946c41cd5>,

 <Addon: 2: Addôn 1aa2878bb3604c31b44f4aba997316cc>]

        got        = []

        response   = <Response status_code=200, "application/json">

        self       = <olympia.lib.es.tests.test_commands.TestIndexCommand testMethod=test_reindexation_starting_from_one_addon>

------------------------------ Captured log call -------------------------------

INFO     z.elasticsearch:reindex.py:79 Returning reindexing group for test_amo_addons-20200602162717

INFO     z.elasticsearch:reindex.py:57 Create the index test_amo_addons-20200602162717, for alias: test_amo_addons

INFO     z.elasticsearch:reindex.py:64 Flagging the database to start the reindexation

INFO     z.elasticsearch:reindex.py:50 Rebuilding aliases with actions: [{'remove': {'index': 'test_amo_addons-20200602162433', 'alias': 'test_amo_addons'}}, {'add': {'index': 'test_amo_addons-20200602162717', 'alias': 'test_amo_addons'}}]

INFO     z.elasticsearch:reindex.py:71 Unflagging the database

INFO     z.elasticsearch:reindex.py:44 Removing indices 'test_amo_addons-20200602162433'
```

**Comment by bobsilverberg** [comment]
I saw a different ES related failure today: https://travis-ci.org/github/mozilla/addons-server/jobs/694310235

That one is on `TestIndexCommand.test_reindexation_with_wipe`. Not sure if we need a separate issue for that, or if the fix for it will be the same as the fix for the one reported here.

**Comment by eviljeff** [comment]
Traceback for theo ther ES test fail:
```
_________________ TestIndexCommand.test_reindexation_with_wipe _________________

src/olympia/lib/es/tests/test_commands.py:172: in test_reindexation_with_wipe

    self._test_reindexation(wipe=True)

        self       = <olympia.lib.es.tests.test_commands.TestIndexCommand testMethod=test_reindexation_with_wipe>

src/olympia/lib/es/tests/test_commands.py:132: in _test_reindexation

    self.refresh()

        ReindexThread = <class 'olympia.lib.es.tests.test_commands.TestIndexCommand._test_reindexation.<locals>.ReindexThread'>

        old_addons_count = 1

        old_indices = [('test_amo_addons_stats_download_counts-20200609080209',

  'test_amo_addons_stats_download_counts'),

 ('test_amo_addons_stats_update_counts-20200609080209',

  'test_amo_addons_stats_update_counts')]

        self       = <olympia.lib.es.tests.test_commands.TestIndexCommand testMethod=test_reindexation_with_wipe>

        t          = <ReindexThread(Thread-3, stopped 139937182476032)>

        wipe       = True

src/olympia/amo/tests/__init__.py:925: in refresh

    cls.es.indices.refresh(settings.ES_INDEXES[index])

        cls        = <class 'olympia.lib.es.tests.test_commands.TestIndexCommand'>

        index      = 'default'

.tox/es/lib/python3.6/site-packages/elasticsearch/client/utils.py:73: in _wrapped

    return func(*args, params=params, **kwargs)

        args       = (<elasticsearch.client.indices.IndicesClient object at 0x7f45ae18a0b8>,

 'test_amo_addons')

        es_query_params = ('allow_no_indices',

 'expand_wildcards',

 'force',

 'ignore_unavailable',

 'operation_threading')

        func       = <function IndicesClient.refresh at 0x7f45b724e840>

        kwargs     = {}

        p          = 'request_timeout'

        params     = {}

.tox/es/lib/python3.6/site-packages/elasticsearch/client/indices.py:56: in refresh

    '_refresh'), params=params)

        index      = 'test_amo_addons'

        params     = {}

        self       = <elasticsearch.client.indices.IndicesClient object at 0x7f45ae18a0b8>

.tox/es/lib/python3.6/site-packages/elasticsearch/transport.py:312: in perform_request

    status, headers, data = connection.perform_request(method, url, params, body, ignore=ignore, timeout=timeout)

        attempt    = 0

        body       = None

        connection = <Urllib3HttpConnection: http://127.0.0.1:9200>

        ignore     = ()

        method     = 'POST'

        params     = {}

        retry      = False

        self       = <elasticsearch.transport.Transport object at 0x7f45ae18a320>

        timeout    = None

        url        = '/test_amo_addons/_refresh'

.tox/es/lib/python3.6/site-packages/elasticsearch/connection/http_urllib3.py:129: in perform_request

    self._raise_error(response.status, raw_data)

        body       = None

        duration   = 0.0020112991333007812

        full_url   = 'http://127.0.0.1:9200/test_amo_addons/_refresh'

        ignore     = ()

        kw         = {}

        method     = 'POST'

        params     = {}

        raw_data   = ('{"error":{"root_cause":[{"type":"index_not_found_exception","reason":"no '

 'such '

 'index","resource.type":"index_or_alias","resource.id":"test_amo_addons","index_uuid":"_na_","index":"test_amo_addons"}],"type":"index_not_found_exception","reason":"no '

 'such '

 'index","resource.type":"index_or_alias","resource.id":"test_amo_addons","index_uuid":"_na_","index":"test_amo_addons"},"status":404}')

        response   = <urllib3.response.HTTPResponse object at 0x7f45aae9e358>

        self       = <Urllib3HttpConnection: http://127.0.0.1:9200>

        start      = 1591689896.9294157

        timeout    = None

        url        = '/test_amo_addons/_refresh'

.tox/es/lib/python3.6/site-packages/elasticsearch/connection/base.py:125: in _raise_error

    raise HTTP_EXCEPTIONS.get(status_code, TransportError)(status_code, error_message, additional_info)

E   elasticsearch.exceptions.NotFoundError: TransportError(404, 'index_not_found_exception', 'no such index')

        additional_info = {'error': {'index': 'test_amo_addons',

           'index_uuid': '_na_',

           'reason': 'no such index',

           'resource.id': 'test_amo_addons',

           'resource.type': 'index_or_alias',

           'root_cause': [{'index': 'test_amo_addons',

                           'index_uuid': '_na_',

                           'reason': 'no such index',

                           'resource.id': 'test_amo_addons',

                           'resource.type': 'index_or_alias',

                           'type': 'index_not_found_exception'}],

           'type': 'index_not_found_exception'},

 'status': 404}

        error_message = 'index_not_found_exception'

        raw_data   = ('{"error":{"root_cause":[{"type":"index_not_found_exception","reason":"no '

 'such '

 'index","resource.type":"index_or_alias","resource.id":"test_amo_addons","index_uuid":"_na_","index":"test_amo_addons"}],"type":"index_not_found_exception","reason":"no '

 'such '

 'index","resource.type":"index_or_alias","resource.id":"test_amo_addons","index_uuid":"_na_","index":"test_amo_addons"},"status":404}')

        self       = <Urllib3HttpConnection: http://127.0.0.1:9200>

        status_code = 404
```

**Comment by diox** [comment]
Ok so, we've got 2 different failures:
- One is that the add-on isn't found in the index. This shouldn't happen, we're refreshing the index before checking!
- The other is that the index isn't found when we're refreshing it before trying to wipe it. This shouldn't happen, the index should have been created when the test started.

Focusing on the first one, here are some notes from our Matrix discussion earlier today:
- The failure appears to be right at the beginning, it's missing the add-on we should have started with, not the ones added after
- We don't see this locally - it must be a timing issue
- We can reproduce by removing the `refresh()` call before checking for the add-on in the index
- Interestingly it looks like the test is flawed, that particular statement is expecting to be testing a search against the old index while we're reindexing on the new one, but it looks like from logging we've already made the switch to the new index!
- `refresh()` call is made on the alias so it shouldn't care, but maybe that's part of the timing issue ? Maybe we could work around this by refreshing with a wildcard ? Maybe, to be extra safe, refresh *both* the alias and any indexes ? Changing `ESTestCase.refresh()` to do `cls.es.indices.refresh(','.join([settings.ES_INDEXES['default'] + '-*', settings.ES_INDEXES['default']]))` or something like that.
